test/k4FWCoreTest/src/components/ExampleFunctionalProducerHist.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2014-2024 Key4hep-Project.
+ *
+ * This file is part of Key4hep.
+ * See https://key4hep.github.io/key4hep-doc/ for further info.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "edm4hep/MCParticleCollection.h"
+
+#include "k4FWCore/Producer.h"
+
+#include "Gaudi/Accumulators/RootHistogram.h"
+#include "GaudiKernel/RndmGenerators.h"
+
+#include <string>
+
+struct ExampleFunctionalProducerHist final : k4FWCore::Producer<edm4hep::MCParticleCollection()> {
```

Comment:
Since we only produce an empty collection here, could we not make this a `Consumer` (or something else that doesn't have an output)?

Reply:
It could take the MCParticles as inputs and then histogram their energy. Or a different datatype for which we already have a functional producer. That would also remove the threadsafety issue with the random numbers(?)